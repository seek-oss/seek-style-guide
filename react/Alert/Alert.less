@import (reference) "~seek-style-guide/theme";

@icon-width: @row-height * 5;
@gutter-adjustment: 100px;
@alert-gutter: @gutter-width * 0.75;

.root {
  position: relative;
}

.alert {
  display: flex;
  align-items: center;
}

.positive {
  color: @sk-positive;
}

.info {
  color: @sk-info;
}

.critical {
  color: @sk-critical;
}

.help {
  color: @sk-help;

  .icon {
    position: relative;
    top: -2px;
  }
}

.text {
  display: inline-block;
  vertical-align: middle;
  flex-grow: 1;
}

.icon {
  vertical-align: middle;
  text-align: center;
  width: @icon-width;
  height: @icon-width;
  flex-shrink: 0;
  margin-right: @alert-gutter;
  @media @mobile {
    vertical-align: top;
  }

  svg {
    width: @icon-width;
    height: @icon-width;
  }

  .primary & {
    opacity: 0.8;
  }
}

.close {
  width: @icon-width;
  height: @icon-width;
  flex-shrink: 0;
  border: none;
  padding: 0;
  background-color: transparent;
  color: currentColor;
  margin-left: @alert-gutter;
  vertical-align: middle;
  cursor: pointer;

  .primary & {
    opacity: 0.8;
  }

  .secondary & {
    color: @sk-black;
    opacity: 0.5;
  }
}
