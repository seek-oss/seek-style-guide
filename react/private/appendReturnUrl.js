export default (url, returnUrl) => {
  const urlSuffix = returnUrl
    ? `?returnUrl=${encodeURIComponent(returnUrl)}`
    : '';

  return `${url}${urlSuffix}`;
};
