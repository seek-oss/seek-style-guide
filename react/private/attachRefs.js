// @flow
export default (...refs: Array<Function>) => (
  ref: ?React$ElementRef<'input'> | ?React$ElementRef<'textarea'>
): void => {
  refs.forEach(callRef => {
    if (typeof callRef === 'function') {
      callRef(ref);
    }
  });
};
