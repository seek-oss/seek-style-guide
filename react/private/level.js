// @flow
export type Level = 'primary' | 'secondary' | 'tertiary';

export const LEVEL: {
  ['PRIMARY' | 'SECONDARY' | 'TERTIARY']: Level
} = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary'
};
