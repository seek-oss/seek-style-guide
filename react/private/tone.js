// @flow
type Tone = 'positive' | 'info' | 'critical' | 'help' | 'neutral';

export const TONE: {
  ['POSITIVE' | 'INFO' | 'CRITICAL' | 'HELP' | 'NEUTRAL']: Tone
} = {
  POSITIVE: 'positive',
  INFO: 'info',
  CRITICAL: 'critical',
  HELP: 'help',
  NEUTRAL: 'neutral'
};
