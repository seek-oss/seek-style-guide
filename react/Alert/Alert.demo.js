import React from 'react';
import PropTypes from 'prop-types';
import { Alert, Card, Section } from 'seek-style-guide/react';
import { TONE } from '../private/tone';
import { LEVEL } from '../private/level';
import * as sketch from './Alert.sketch';

const Container = ({ component: DemoComponent, componentProps }) => (
  <Card style={{ width: '500px' }}>
    <Section>
      <DemoComponent {...componentProps} />
    </Section>
  </Card>
);

Container.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

export default {
  route: '/alert',
  title: 'Alert',
  category: 'Form',
  component: Alert,
  container: Container,
  sketch,
  initialProps: {
    tone: TONE.POSITIVE,
    level: LEVEL.PRIMARY,
    message: "I'm some timely text that helps the user understand something",
    hideIcon: false,
    onClose: null
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Hide icon',
          transformProps: ({ hideIcon, ...props }) => ({
            ...props,
            hideIcon: !hideIcon
          })
        },
        {
          label: 'Close button / handler',
          transformProps: ({ onClose, ...props }) => ({
            ...props,
            onClose: () => console.log('On close handler called')
          })
        },
        {
          label: 'Pullout',
          transformProps: ({ pullout, ...props }) => ({
            ...props,
            pullout: !pullout
          })
        }
      ]
    },
    {
      label: 'Tone',
      type: 'radio',
      states: [
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            tone: TONE.POSITIVE
          })
        },
        {
          label: 'Info',
          transformProps: props => ({
            ...props,
            tone: TONE.INFO
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            tone: TONE.CRITICAL
          })
        },
        {
          label: 'Help',
          transformProps: props => ({
            ...props,
            tone: TONE.HELP
          })
        }
      ]
    },
    {
      label: 'Level',
      type: 'radio',
      states: [
        {
          label: 'Primary',
          transformProps: props => ({
            ...props,
            level: LEVEL.PRIMARY
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            level: LEVEL.SECONDARY
          })
        },
        {
          label: 'Tertiary',
          transformProps: props => ({
            ...props,
            level: LEVEL.TERTIARY
          })
        }
      ]
    }
  ]
};
