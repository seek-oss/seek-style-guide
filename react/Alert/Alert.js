import styles from './Alert.less';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import omit from 'lodash/omit';

import Section from '../Section/Section';
import Text from '../Text/Text';
import TickCircleIcon from '../TickCircleIcon/TickCircleIcon';
import InfoIcon from '../InfoIcon/InfoIcon';
import CriticalIcon from '../CriticalIcon/CriticalIcon';
import HelpIcon from '../HelpIcon/HelpIcon';
import CrossIcon from '../CrossIcon/CrossIcon';

import { TONE } from '../private/tone';
import { LEVEL } from '../private/level';

const ICONS = {
  [TONE.POSITIVE]: TickCircleIcon,
  [TONE.INFO]: InfoIcon,
  [TONE.CRITICAL]: CriticalIcon,
  [TONE.HELP]: HelpIcon
};

export default class Alert extends Component {
  static displayName = 'Alert';

  static propTypes = {
    tone: PropTypes.oneOf([TONE.POSITIVE, TONE.INFO, TONE.CRITICAL, TONE.HELP])
      .isRequired,
    level: PropTypes.oneOf([LEVEL.PRIMARY, LEVEL.SECONDARY, LEVEL.TERTIARY])
      .isRequired,
    message: PropTypes.node.isRequired,
    pullout: PropTypes.bool,
    hideIcon: PropTypes.bool,
    onClose: PropTypes.func
  };

  static defaultProps = {
    hideIcon: false,
    pullout: false,
    onClose: null
  };

  handleClose = event => this.props.onClose(event);

  renderContents = () => {
    const { tone, message, hideIcon, onClose } = this.props;

    const Icon = tone ? ICONS[tone] : null;

    return (
      <div className={styles.alert}>
        {!hideIcon && Icon && <Icon className={styles.icon} />}
        <div className={styles.text}>
          <Text raw baseline={false}>
            {message}
          </Text>
        </div>
        {onClose && (
          <button className={styles.close} onClick={this.handleClose}>
            <CrossIcon />
          </button>
        )}
      </div>
    );
  };

  render() {
    const {
      hideIcon,
      onClose,
      tone,
      level,
      pullout,
      ...restProps
    } = this.props;

    const additionalProps = omit(restProps, 'message');

    const isTertiary = level === LEVEL.TERTIARY;

    const rootClasses = classnames({
      [styles.root]: true,
      [styles[tone]]: tone && isTertiary
    });

    return isTertiary ? (
      <div className={rootClasses} {...additionalProps}>
        {this.renderContents()}
      </div>
    ) : (
      <Section
        tone={tone}
        level={level}
        pullout={pullout}
        className={rootClasses}
        {...additionalProps}
      >
        {this.renderContents()}
      </Section>
    );
  }
}
