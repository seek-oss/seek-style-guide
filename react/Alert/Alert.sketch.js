import React from 'react';
import Alert from './Alert';
import SketchFieldContainer from '../private/SketchFieldContainer/SketchFieldContainer';

const noop = () => {};

export const symbols = {
  'Alert/1. Positive/1. Primary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="positive" level="primary" />
    </SketchFieldContainer>
  ),
  'Alert/1. Positive/1. Primary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="positive" level="primary" onClose={noop} />
    </SketchFieldContainer>
  ),
  'Alert/1. Positive/2. Secondary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="positive" level="secondary" />
    </SketchFieldContainer>
  ),
  'Alert/1. Positive/2. Secondary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert
        message="Message"
        tone="positive"
        level="secondary"
        onClose={noop}
      />
    </SketchFieldContainer>
  ),
  'Alert/1. Positive/3. Tertiary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="positive" level="tertiary" />
    </SketchFieldContainer>
  ),
  'Alert/1. Positive/3. Tertiary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert
        message="Message"
        tone="positive"
        level="tertiary"
        onClose={noop}
      />
    </SketchFieldContainer>
  ),

  'Alert/2. Critical/1. Primary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="critical" level="primary" />
    </SketchFieldContainer>
  ),
  'Alert/2. Critical/1. Primary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="critical" level="primary" onClose={noop} />
    </SketchFieldContainer>
  ),
  'Alert/2. Critical/2. Tertiary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="critical" level="tertiary" />
    </SketchFieldContainer>
  ),
  'Alert/2. Critical/2. Tertiary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert
        message="Message"
        tone="critical"
        level="tertiary"
        onClose={noop}
      />
    </SketchFieldContainer>
  ),

  'Alert/3. Info/1. Primary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="primary" />
    </SketchFieldContainer>
  ),
  'Alert/3. Info/1. Primary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="primary" onClose={noop} />
    </SketchFieldContainer>
  ),
  'Alert/3. Info/2. Secondary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="secondary" />
    </SketchFieldContainer>
  ),
  'Alert/3. Info/2. Secondary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="secondary" onClose={noop} />
    </SketchFieldContainer>
  ),
  'Alert/3. Info/3. Tertiary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="tertiary" />
    </SketchFieldContainer>
  ),
  'Alert/3. Info/3. Tertiary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="info" level="tertiary" onClose={noop} />
    </SketchFieldContainer>
  ),

  'Alert/4. Help/1. Primary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="help" level="primary" />
    </SketchFieldContainer>
  ),
  'Alert/4. Help/1. Primary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="help" level="primary" onClose={noop} />
    </SketchFieldContainer>
  ),
  'Alert/4. Help/2. Tertiary/1. Standard': (
    <SketchFieldContainer>
      <Alert message="Message" tone="help" level="tertiary" />
    </SketchFieldContainer>
  ),
  'Alert/4. Help/2. Tertiary/2. With Close Button': (
    <SketchFieldContainer>
      <Alert message="Message" tone="help" level="tertiary" onClose={noop} />
    </SketchFieldContainer>
  )
};
