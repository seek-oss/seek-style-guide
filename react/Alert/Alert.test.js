import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';
import { TONE } from '../private/tone';
import { LEVEL } from '../private/level';

const renderAlert = props =>
  shallow(
    <Alert
      tone={TONE.POSITIVE}
      level={LEVEL.PRIMARY}
      message="Test message"
      {...props}
    />
  );

describe('Alert:', () => {
  describe('types:', () => {
    it('should render positive alerts', () => {
      const alert = renderAlert({ tone: TONE.POSITIVE });
      expect(alert).toMatchSnapshot();
    });

    it('should render info alerts', () => {
      const alert = renderAlert({ tone: TONE.INFO });
      expect(alert).toMatchSnapshot();
    });

    it('should render critical alerts', () => {
      const alert = renderAlert({ tone: TONE.CRITICAL });
      expect(alert).toMatchSnapshot();
    });

    it('should render help alerts', () => {
      const alert = renderAlert({ tone: TONE.HELP });
      expect(alert).toMatchSnapshot();
    });
  });

  describe('levels:', () => {
    it('should render primary alerts', () => {
      const alert = renderAlert({ level: LEVEL.PRIMARY });
      expect(alert).toMatchSnapshot();
    });

    it('should render secondary alerts', () => {
      const alert = renderAlert({ level: LEVEL.SECONDARY });
      expect(alert).toMatchSnapshot();
    });

    it('should render tertiary alerts', () => {
      const alert = renderAlert({ level: LEVEL.TERTIARY });
      expect(alert).toMatchSnapshot();
    });
  });

  it('should render the pullout style', () => {
    const alert = renderAlert({ pullout: true });
    expect(alert).toMatchSnapshot();
  });

  it('should render with a hidden icon', () => {
    const alert = renderAlert({ hideIcon: true });
    expect(alert).toMatchSnapshot();
  });

  it('should render a close button', () => {
    const onClose = jest.fn();
    const alert = renderAlert({ onClose });
    expect(alert).toMatchSnapshot();
  });

  it('should call a passed in onClose function when the button fires a click event', () => {
    const onClose = jest.fn();
    const alert = renderAlert({ onClose });
    const event = { someEventData: true };

    alert.find('button').simulate('click', event);
    expect(onClose).toBeCalledWith(event);
  });

  it('should pass through additional props', () => {
    const alert = renderAlert({ 'custom-prop': true });
    expect(alert).toMatchSnapshot();
  });

  it('will accept a node in a the message prop', () => {
    const alert = renderAlert({ message: <span>I'm a node</span> });
    expect(alert).toMatchSnapshot();
  });
});
