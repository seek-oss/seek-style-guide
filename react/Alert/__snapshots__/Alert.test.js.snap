// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert: levels: should render primary alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: levels: should render secondary alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="secondary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: levels: should render tertiary alerts 1`] = `
<div
  className="Alert__root Alert__positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</div>
`;

exports[`Alert: should pass through additional props 1`] = `
<Section
  className="Alert__root"
  custom-prop={true}
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: should render a close button 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
    <button
      className="Alert__close"
      onClick={[Function]}
    >
      <CrossIcon />
    </button>
  </div>
</Section>
`;

exports[`Alert: should render the pullout style 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={true}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: should render with a hidden icon 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: types: should render critical alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="critical"
>
  <div
    className="Alert__alert"
  >
    <CriticalIcon
      className="Alert__icon"
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: types: should render help alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="help"
>
  <div
    className="Alert__alert"
  >
    <HelpIcon
      className="Alert__icon"
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: types: should render info alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="info"
>
  <div
    className="Alert__alert"
  >
    <InfoIcon
      className="Alert__icon"
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: types: should render positive alerts 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        Test message
      </Text>
    </div>
  </div>
</Section>
`;

exports[`Alert: will accept a node in a the message prop 1`] = `
<Section
  className="Alert__root"
  header={false}
  level="primary"
  pullout={false}
  slim={false}
  tone="positive"
>
  <div
    className="Alert__alert"
  >
    <TickCircleIcon
      className="Alert__icon"
      filled={false}
    />
    <div
      className="Alert__text"
    >
      <Text
        baseline={false}
        raw={true}
      >
        <span>
          I'm a node
        </span>
      </Text>
    </div>
  </div>
</Section>
`;
